src/randoop/JunitFileWriter.java

```diff
@@ -195,58 +183,86 @@ public File writeDriverFile(String driverClassName) {
     return junitTestSuites;
   }
   
-  public static File writeDriverFile(File dir, String packageName, String driverClassName,
-      List<String> junitTestSuiteNames) {
-    File file = new File(dir, driverClassName + ".java");
+  /*
+   * writes JUnit4 suite based on list of class names
+   */
+  public File writeSuiteFile (List<String> testClassNames) {
+    File dir = this.getDir();
+    String suiteClassName = masterTestClassName;
+    File file = new File(dir, suiteClassName+".java");
     PrintStream out = createTextOutputStream(file);
     try {
-      outputPackageName(out, packageName);
-      out.println("import junit.framework.*;");
-      out.println("import junit.textui.*;");
-      out.println("");
-      out.println("public class " + driverClassName + " extends TestCase {");
-      out.println("");
-      if (GenInputsAbstract.junit_reflection_allowed) {
-      out.println("  public static void main(String[] args) {");
-      } else {
-      out.println("  public static void main(String[] args) throws Throwable {");
+      outputPackageName(out,packageName);
+      
+      out.println();
+      out.println("import org.junit.runner.RunWith;");
+      out.println("import org.junit.runners.Suite;");
+      out.println();
+      out.println("@RunWith(Suite.class)");
+      out.println("@Suite.SuiteClasses({");
+     
+      Iterator<String> testIterator = testClassNames.iterator();
```

Comment:
I would replace the next 7 lines by a call to join.  Theri is a [join](http://types.cs.washington.edu/plume-lib/api/plume/UtilMDE.html#join-java.util.List-java.lang.String-), though maybe a more canonical one than in plume-lib exists.
I would also put the commas at the ends of lines rather than at the beginnings of lines.